test/change.test.js

```diff
@@ -219,6 +272,92 @@ describe('Change', function() {
     });
   });
 
+  describe('change.rectify - promise variant', function() {
+    var change;
+    beforeEach(function(done) {
+      Change.findOrCreateChange(this.modelName, this.modelId)
+        .then(function(ch) {
+          change = ch;
+          done();
+        })
+        .catch(done);
+    });
+
+    it('should create a new change with the correct revision', function(done) {
+      var test = this;
+      change.rectify()
+        .then(function(ch) {
+          assert.equal(ch.rev, test.revisionForModel);
+          done();
+        })
+        .catch(done);
+    });
+
+    // This test is a low-level equivalent of the test in replication.test.js
+    // called "replicates multiple updates within the same CP"
+    it('should merge updates within the same checkpoint', function(done) {
```

Comment:
Please remove this test, it's a complex one that checks internal behaviour of checkpoints, not the public API. It's better to not duplicate a complex code like this, plus it's very unlikely this would fail just because of a promise API.